### DISCUSSION TITLE
"indicator" tile ?

### DISCUSSION THREAD

**Discussion by Sill1000** [HomeSpan/HomeSpan#531] [Q&A]
Hi,
 
Does HK/HS
![Capture d’écran 2023-03-16 à 18 26 51](https://user-images.githubusercontent.com/88202590/225703689-df733073-0f06-42a4-93d5-981d40c17867.jpg)
![Capture d’écran 2023-03-16 à 18 26 44](https://user-images.githubusercontent.com/88202590/225703693-8650a067-2beb-4d57-b171-7146ef9db042.jpg)
![Capture d’écran 2023-03-16 à 18 26 33](https://user-images.githubusercontent.com/88202590/225703694-ba4c5d87-5875-4b18-8b1c-327d95449b03.jpg)
 have the possibility to have an "indicatior tile" in which you can change the text according to the status of the input pin(s) ? See my example.

Thanks

**Comment by HomeSpan** [comment]
Not that I am aware of.

**Comment by Sill1000** [comment]
...or an another way to display a test on the Homekit App ?

Thanks

**Comment by Sill1000** [comment]
Maybe the solution is using the Service::InputSource(); 
I will try it

**Comment by Sill1000** [comment]
I'm trying the InputSource and for testing I would like that every 5 seconds the menu is changing automatically (effect1<->effect2)

But I'm fully lost with C++. How to write this into the loop???
Could anyone help me? Thanks

.ino:
```
homeSpan.begin(Category::Bridges, "Test");  // initializes a HomeSpan device named "HomeSpan Lightbulb" with Category set to Bridge

  new SpanAccessory();     
  new Service::AccessoryInformation();
  new Characteristic::Identify();

    
    new SpanAccessory();  
    new Service::AccessoryInformation();
    new Characteristic::Identify();
    new Characteristic::Name("Menu");
    new DEV_test();
```


.h:

```
struct DEV_test : Service::LightBulb {       

  DEV_test(): Service::LightBulb() {
    new Characteristic::On();

    SpanService *effect1 = new Service::InputSource();
    new Characteristic::ConfiguredName("effect 1");
    new Characteristic::Identifier(1);

    SpanService *effect2 = new Service::InputSource();
    new Characteristic::ConfiguredName("effect 2");
    new Characteristic::Identifier(2);

    (new Service::Television())
    ->addLink(effect1)
    ->addLink(effect2);
    new Characteristic::Active(1);
    new Characteristic::ActiveIdentifier(1);
  }

  boolean update()
  {
    LOG1("Button power update");
    LOG1("\n");
    return (true);
  }

  void loop()
  {
    /*   if (effect->timeVal() > 5000)
       {
         Serial.println("change to Menu 2");
         ->addLink(effect2);
       }
       else
       {
         Serial.println("change to Menu 1");
         ->addLink(effect1);
       }*/
  }
};
```
